src/NexusMods.Networking.GOG/Client.cs

```diff
@@ -241,6 +242,106 @@ public bool TryGetAuthInfo(out AuthInfo.ReadOnly authInfo)
+            if (productResponse is null) throw new JsonException($"Failed to deserialize product response for {productId}");
+            var result = productResponse.Downloads.Installers.Select(installer =>
+            {
+                var downloadLink = new Uri(installer.Files[0].Downlink);
+                var os = installer.OS switch
+                {
+                    "linux" => OSPlatform.Linux,
+                    "osx" or "mac" => OSPlatform.OSX,
+                    "windows" => OSPlatform.Windows,
+                    _ => OSPlatform.Create(installer.OS),
+                };
+
+                return (os, downloadLink);
+            }).ToArray();
+
+            return result;
+        }, cancellationToken: token);
+    }
+
+    public async ValueTask DownloadInstallerArchive((OSPlatform OS, Uri DownloadLink) installerInfo, Stream output, CancellationToken token)
+    {
+        if (installerInfo.OS != OSPlatform.Linux) throw new NotSupportedException();
+        // NOTE(erri120): Linux installers are downloaded as shell scripts.
+        // The shell script is a self-extractable archive created using makeself (https://makeself.io/)
+        // A startup script is configured to run a mojosetup script (https://icculus.org/mojosetup/)
+        // We don't really care about any of this since we can just get the game contents directly by finding
+        // the ZIP archive embedded in the downloaded file and only extracting the game files.
+
+        await _pipeline.ExecuteAsync(async cancellationToken =>
+        {
+            Uri downloadLink;
+
+            using (var msg = await CreateMessage(installerInfo.DownloadLink, cancellationToken))
+            using (var response = await _client.SendAsync(msg, cancellationToken: cancellationToken))
+            {
+                var installerResponse = await response.Content.ReadFromJsonAsync<InstallerResponse>(_jsonSerializerOptions, cancellationToken: cancellationToken);
+                if (installerResponse is null) throw new JsonException("Failed to get installer");
+                downloadLink = new Uri(installerResponse.DownloadLink);
+            }
+
+            using (var msg = await CreateMessage(downloadLink, cancellationToken))
+            using (var response = await _client.SendAsync(msg, cancellationToken: cancellationToken))
+            {
+                await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken: cancellationToken);
+                await CopyArchive(stream, output, cancellationToken);
+            }
+        }, cancellationToken: token);
+    }
+
+    private static async ValueTask CopyArchive(Stream input, Stream output, CancellationToken cancellationToken)
+    {
+        var bufferSize = Size.MB;
+        var buffer = new Memory<byte>(array: new byte[bufferSize.Value]);
```

Comment:
We could probably skip zero initialization here of `new byte[`, but it's not too important in the grand scheme of things.